components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -522,6 +529,14 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                 LOG.debug("Retrieved the User Information:" + jwtAttributeMap);
             }
 
+            String nonce = (String) jwtAttributeMap.get(NONCE);
+            if (nonce == null) {
+                LOG.warn("OIDC provider does not support nonce claim in id_token.");
```

Comment:
Should this be a warn log? if its good to have a warn log then i would suggest to have a config at some level to disable the nonce validation as if the IS is connected to a OIDC provider without nonce validation there will be a load of warn logs.

We can have the following options
1- Turn the warn log into a debug log (if it isn't an issue for the requirement)
2- Provide a config to disable nonce validation

Reply:
Selected the 1st option as the 2nd option include much effort on UI related improvements. Also there is PKCE based support for the moment and wanted to provide the Nonce based protection out of the box as nonce claim is defined in OIDC specs, hence most of the IDPs should support for this prevention mechanism.
Fixed by - https://github.com/wso2-extensions/identity-outbound-auth-oidc/pull/139/commits/8bb6ffe6e5640808f85b2ef23c299fbd0be747d5